### PR TITLE
Capture database type in config

### PR DESCRIPTION
**Pull request by bleskes**
Today we allow user to specify a database type as part of their database path when opening duckdb (example: `.open sqllite:path_to_db`). This is picked up at initialization time and is used to route the opening command to the right extension. 

This works fine, but the information the database type is lost. While not a big deal, that means that if you wanted to know what the type was of the db the user opened, you have to jump through some hoops to get to it (get the default database, resolve the catalog, cast it to classes etc.). 

This PR adds an explicit configuration option and populates it at initialization time following the same logic of today.

### CODE REVIEW

**Review by Mytherin — COMMENTED** [review]
Thanks for the PR! LGTM - one minor comment

File: src/include/duckdb/main/database.hpp [hunk 1]
```diff
@@ -54,13 +54,13 @@ class DatabaseInstance : public std::enable_shared_from_this<DatabaseInstance> {
 
 	DUCKDB_API bool TryGetCurrentSetting(const std::string &key, Value &result);
 
-	//! Get the database extension type from a given path
-	string ExtractDatabaseType(string &path);
+	//! Get the database extension type from a given path, returning a pair of <extension type, rest of path>
+	pair<string, string> ExtractDatabaseType(const string &path);
```
[the remark this answers is not in the document]
Comment (bleskes) [reply src/include/duckdb/main/database.hpp]: sure thing. Updated.

[the remark this answers is not in the document]
Comment (bleskes) [reply src/include/duckdb/main/database.hpp]: @Mytherin this might be more than what you were thinking of, so LMK if you want something else.

**Comment by bleskes** [comment]
The failing test in Vector Sizes passes locally for me and I can't see how it's related. Any advice?

```
~/src/md/duckdb/build/debug (md_type) » test/unittest test/sql/pivot/pivot_example.test                                                                                             boaz@boaz-air
Filters: test/sql/pivot/pivot_example.test
[1/1] (100%): test/sql/pivot/pivot_example.test
===============================================================================
All tests passed (202 assertions in 1 test case)
```

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**